lib/providers/map_data.dart

```diff
@@ -1,26 +1,24 @@
 import 'dart:async';
 import 'package:concordia_navigation/models/itinerary.dart';
-import 'package:concordia_navigation/services/size_config.dart';
 import 'package:flutter/foundation.dart';
 import 'package:google_maps_flutter/google_maps_flutter.dart';
 import 'package:flutter/material.dart';
+import 'package:sliding_up_panel/sliding_up_panel.dart';
 
 ///Observer Pattern
 ///Handles all the data related to the map, listens to changes and notifies listeners.
 class MapData extends ChangeNotifier {
   Completer<GoogleMapController> _completer = Completer();
+  PanelController panelController = new PanelController();
```

Comment:
Just as an fyi for me, whats a panel controller?

Reply:
The panel controller is the controller class for the SlideUpPanel widget that contains the DirectionsDrawer.

With it, you can do things like open or hide the panel. I put it in the provider to make it accessible throughout the app. [here's the docs for it](https://pub.dev/documentation/sliding_up_panel/latest/sliding_up_panel/PanelController-class.html).

This way, when an element wants to close the directions widget (it is pretty obtrusive) they can call `Provider.of<MapData>(context, listen: false).panelController.show()`.